library/src/main/scala/scala/scalajs/js/Union.scala

```diff
@@ -66,6 +79,14 @@ object | { // scalastyle:ignore
   implicit def from[A, B1, B2](a: A)(implicit ev: Evidence[A, B1 | B2]): B1 | B2 =
     a.asInstanceOf[B1 | B2]
 
+  /** Upcast `F[A]` to `F[B]`.
+   *
+   *  This needs evidence that `F[A] <: F[B]`.
+   */
+  implicit def fromTypeConstructor[F[_], A, B](a: F[A])(
+      implicit ev: Evidence[F[A], F[B]]): F[B] =
```

Comment:
So just to double check: You allow conversion to *any* type in the constructor so that, for example, the following apparent subtyping relationship holds: `List[List[Int]] <: List[List[Int | String]]`.